test/controllers/minio.py

```diff
@@ -0,0 +1,65 @@
+import os
+from pathlib import Path
+import shutil
+import subprocess
+import tempfile
+import time
+
+from conftest import find_free_port
+
+class MinioController:
+    # ported from https://github.com/kbase/java_test_utilities/blob/develop/src/main/java/us/kbase/testutils/controllers/minio/MinioController.java
+    
+    def __init__(self, minioexe: Path, access_key: str, secret_key: str, root_temp_dir: Path):
+        root_temp_dir = root_temp_dir.absolute()
+        root_temp_dir.mkdir(parents=True, exist_ok=True)
+        tdir = tempfile.mkdtemp(dir=root_temp_dir, prefix="MinioController-")
+        self.tempdir = root_temp_dir / tdir
+        datadir = self.tempdir / "data"
+        datadir.mkdir()
+        
+        self.port = find_free_port()
+        
+        logfile = self.tempdir / "minio_server.log"
+        
+        self._logfiledescriptor = open(logfile, "w")
+        
+        cmd = [
+                minioexe,
+                "server",
+                "--compat",
+                "--address", f'localhost:{self.port}',
+                # --console-address doesn't seem to work on the 2021-4 Minio distro, maybe too old
+                str(datadir)
+        ]
+        env = dict(os.environ)
+        env["MINIO_ACCESS_KEY"] = access_key
+        env["MINIO_SECRET_KEY"] = secret_key
```

Comment:
I'm not entirely certain about the use case for the MinIO controller. However, it seems like we're setting environment variables through a function call. Each call to the function can pass different access keys and secret keys, which would result in the environment variables being updated. Is that the behavior we are looking for? Should we set those keys via config file instead?

Reply:
I"m not sure I understand - each creation of the controller can specify different keys, but the global environment is not affected due to the `dict(os.environ)` line. Each invocation of Minio gets its own unique environment, so they can't conflict.

If that doesn't answer your concern, can you elaboate, because I don't understand it in that case.

Reply:
oh I see it now. Cool. Sounds good.